java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java

```diff
@@ -65,23 +66,45 @@ public void visitNode(Tree tree) {
   }
 
   private static ModifierTree getFirstBadlyOrdered(ModifiersTree modifiersTree) {
-    int modifierIndex = -1;
+    ListIterator<ModifierTree> modifiersIterator = modifiersTree.listIterator();
+    skipAnnotations(modifiersIterator);
```

Comment:
It should be way simpler to generate a new list of `modifierTree` from the provided list, but without all the annotations at its start and at its end of the list. and then use the same logic as before.